### PR TITLE
Feat: Create Elasticsearch `elasticsearch_protocol` Setting

### PR DESCRIPTION
**Pull request by voxparcxls**
Elasticsearch Endpoint -- Protocol and Host configuration modified

`elasticsearch_protocol=HTTP`

`elasticsearch_host=localhost`

* CWSInstaller will check for user provided `elasticsearch_protocol` through both "interactive mode" and "configured properties" installation.
 
  - `elasticsearch_protocol` Protocol options: `HTTP` or `HTTPS` (checks for other cases i.e. 'http', 'https', 'HTTP', 'HTTPS')
  - If `elasticsearch_host` is set to a URL with its protocol included (ex: "http://localhost"), CWS installation will default to using `elasticsearch_protocol` and will display WARNING during validation.
  
Elasticsearch WARNING:

Screen Shot 2022-04-13 at 4 02 05 PM

### CODE REVIEW

File: install/example-cws-configuration.properties [hunk 1]
```diff
@@ -116,6 +116,9 @@ cws_notification_emails=[AdminEmailAddresses(comma separated)]
 # This allows CWS to use a user provided Elasticsearch service.
 # If you wish to use an unsecured Elasticsearch host, you may do so
 # by specifying elasticsearch_use_auth=n below.
+# Also specify if your provided Elasticsearch is encrypted(https) or encrypted(http)
+# by setting elasticsearch_protocol to https or http.
+elasticsearch_protocol=[YourElasticsearchProtocol]
```
Comment (jamesfwood) [inline install/example-cws-configuration.properties:121]: Add the choices.  so they can see the possible choices

Comment (jamesfwood) [reply install/example-cws-configuration.properties:121]: 'http://' or 'https://'

Comment (jamesfwood) [reply install/example-cws-configuration.properties:121]: http or https works too

Comment (jamesfwood) [reply install/example-cws-configuration.properties:121]: Make the examples and documentation with "HTTP" or "HTTPS" only

File: cws-installer/src/main/java/jpl/cws/task/CwsInstaller.java [hunk 2]
```diff
@@ -1064,33 +1066,90 @@ private static void setupLogstash() {
 
 	private static void setupElasticsearch() {
```
Comment (jamesfwood) [inline cws-installer/src/main/java/jpl/cws/task/CwsInstaller.java:1068]: Make sure curl is done after the URL is checked

File: install/docker/worker-ls/cws-logstash.conf [hunk 3]
```diff
@@ -53,5 +53,5 @@ filter {
 }
 
 output {
-  elasticsearch { hosts => ["__ES_HOST__:9200"] }
+  elasticsearch { hosts => ["__ES_PROTOCOL__://__ES_HOST__:9200"] }
```
Comment (jamesfwood) [inline install/docker/worker-ls/cws-logstash.conf:56]: make sure to strip out protocol in __ES_HOST__

File: README.md [hunk 4]
```diff
@@ -103,7 +105,8 @@ ADMIN_LAST="{last}"
 ADMIN_EMAIL="{email}"
 
 # ES config
-ES_HOST="http://localhost"
+ES_PROTOCOL="http://"  # options: 'http://' or 'https://'
```
Comment (jamesfwood) [inline README.md:108]: Make this:  http or https

Comment (jamesfwood) [reply README.md:108]: or HTTP
or HTTPS

File: cws-installer/src/main/java/jpl/cws/task/CwsInstaller.java [hunk 5]
```diff
@@ -169,7 +169,10 @@ public class CwsInstaller {
 	private static String cws_security_filter_class;
 	private static String startup_autoregister_process_defs;
 	private static String cws_token_expiration_hours;
+	private static String elasticsearch_protocol;
+	private static String elasticsearch_protocol_temp;
```
Comment (jamesfwood) [inline cws-installer/src/main/java/jpl/cws/task/CwsInstaller.java:173]: change temp to initial

File: cws-service/src/main/java/jpl/cws/controller/RestService.java [hunk 6]
```diff
@@ -405,7 +406,7 @@ private String doDeployProcessDefinition(MultipartFile file) {
 	 * @return fully constructed elasticsearch URL string
 	 */
 	private String constructElasticsearchUrl(String subPath) {
-		String urlString = elasticsearchHostname + ":" + elasticsearchPort + subPath;
+		String urlString = elasticsearchProtocolname + elasticsearchHostname + ":" + elasticsearchPort + subPath;
```
Comment (jamesfwood) [inline cws-service/src/main/java/jpl/cws/controller/RestService.java:409]: change name to Name.  camel case

Comment (jamesfwood) [reply cws-service/src/main/java/jpl/cws/controller/RestService.java:409]: elasticsearchProtocolname => elasticsearchProtocolName

**Comment by voxparcxls** [comment]
Elasticsearch Protocol and Host are tested in both Dev and User configurations.

* Changed variable names: `elasticsearch_protocol_init`, `elasticsearch_host_init`, `elasticsearchProtocolName`
* Checked for lower and upper case protocol input
* Tested in 'Interactive' mode and configuration presets


Ready for merge.

File: install/example-cws-configuration.properties (hunk 1 above)
Comment (voxparcxls) [reply install/example-cws-configuration.properties:125]: Documentation and Interactive mode prompts display uppercase HTTP/HTTPS

File: cws-installer/src/main/java/jpl/cws/task/CwsInstaller.java (hunk 2 above)
Comment (voxparcxls) [reply cws-installer/src/main/java/jpl/cws/task/CwsInstaller.java:1069]: https://github.com/NASA-AMMOS/common-workflow-service/blob/83f251dcb3c79ed16351d77d1739928a314b9cac/cws-installer/src/main/java/jpl/cws/task/CwsInstaller.java#L1934

File: cws-installer/src/main/java/jpl/cws/task/CwsInstaller.java (hunk 5 above)
Comment (voxparcxls) [reply cws-installer/src/main/java/jpl/cws/task/CwsInstaller.java]: Changed to:  `elasticsearch_protocol_init`, `elasticsearch_host_init`

**Review by jamesfwood — APPROVED** [review]
Ok looks good.  Thanks!

**Pull request merged**